### DISCUSSION TITLE
Decide which versioning scheme to use

### DISCUSSION THREAD

**Discussion by phaer** [mysticaltech/terraform-hcloud-kube-hetzner#84] [Ideas]
Recently, the master branch has been quite unstable and required users to either stay on an unsupported, older commit of kube-hetzner or to re-provision their whole cluster. 

I believe that the time would be right to agree on a versioning scheme and implement at least a minimum of a release process to communicate breaking changes more clearly.  

My proposal would be to just use https://semver.org/ and start to define a process after which we could release a 1.0.0 (or 0.1.0 if you prefer ;). Ideally we would end up with a git tag, an auto-generated github release & a ready-to-use module published on registry.terraform.io. 

I also started a GitHub project regarding the whole thing, you can find it linked in the sidebar of this issue or at https://github.com/orgs/kube-hetzner/projects/1

eager to hear what @mysticaltech, @mnencia and others are thinking!

**Comment by mysticaltech** [comment]
@phaer You are right and have been feeling bad about that too lately. Have realized too how users are confused. Thanks for taking this initiative. Definitly, semver is the way to go IMHO. Also, thanks for the project, that will get us more organized too :)
